File: RESTFul/ADS/PHP/app1/lib/ADS/Category.php
<?php
namespace Att\Api\ADS;

/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4 */

/**
 * ADS Library
 * 
 * PHP version 5.4+
 * 
 * LICENSE: Licensed by AT&T under the 'Software Development Kit Tools 
 * Agreement.' 2013. 
 * TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTIONS:
 * http://developer.att.com/sdk_agreement/
 *
 * Copyright 2013 AT&T Intellectual Property. All rights reserved.
 * For more information contact developer.support@att.com
 * 
 * @category  API
 * @package   ADS 
 * @author    pk9069
 * @copyright 2013 AT&T Intellectual Property
 * @license   http://developer.att.com/sdk_agreement AT&amp;T License
 * @link      http://developer.att.com
 */

/**
 * Contains <i>Category</i> constants.
 *
 * @category API
 * @package  ADS
 * @author   pk9069
 * @license  http://developer.att.com/sdk_agreement AT&amp;T License
 * @version  Release: @package_version@ 
 * @link     https://developer.att.com/docs/apis/rest/1/Advertising
 */
final class Category
{
    /**
     * Disallow instances.
     */
    private function __construct() 
    {
    }

    const AUTO = "auto";
    const BUSINESS = "business";
    const FINANCE = "finance";
    const CHAT = "chat";
    const COMMUNITY = "community";
    const SOCIAL = "social";
    const PERSONALS = "personals";
    const COMMUNICATION = "communication";
    const TECHNOLOGY = "technology";
    const GAMES = "games";
    const HEALTH = "health";
    const MEDICAL = "medical";
    const MAPS = "maps";
    const LOCAL = "local";
    const ENTERTAINMENT = "entertainment";
    const MOVIES = "movies";
    const TV = "tv";
    const MUSIC = "music";
    const PHOTOS = "photos";
    const VIDEO = "video";
    const NEWS = "news";
    const WEATHER = "weather";
    const SHOPPING = "shopping";
    const SPORTS = "sports";
    const TOOLS = "tools";
    const TRAVEL = "travel";
    const OTHER = "other";
}

File: RESTFul/ADS/Java/app1/src/main/java/com/att/api/ads/service/Category.java
/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4 */

/*
 * Copyright 2014 AT&T
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.att.api.ads.service;

public enum Category {
    AUTO("auto"), BUSINESS("business"), FINANCE("finance"), CHAT("chat"),
    COMMUNITY("community"), SOCIAL("social"), PERSONALS("personals"),
    COMMUNICATION("communication"), TECHNOLOGY("technology"), GAMES("games"),
    HEALTH("health"), MEDICAL("medical"), MAPS("maps"), LOCAL("local"),
    ENTERTAINMENT("entertainment"), MOVIES("movies"), TV("tv"), MUSIC("music"),
    PHOTOS("photos"), VIDEO("video"), NEWS("news"), WEATHER("weather"),
    SPORTS("sports"), SHOPPING("shopping"), TOOLS("tools"), TRAVEL("travel"),
    OTHER("other");

    private final String val;

    private Category(String val) {
        this.val = val;
    }

    public String getString() {
        return this.val;
    }

    public static Category fromString(final String str) {
        Category[] categories = Category.values();
        for (Category c : categories) {
            if (c.getString().equals(str)) {
                return c;
            }
        }

        return null;
    }

    public static String[] stringValues() {
        Category[] categories = Category.values();
        String[] strs = new String[categories.length];

        for (int i = 0; i < categories.length; ++i) {
            strs[i] = categories[i].getString();
        }

        return strs;
    }
}
